ion/ionops.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Quantum gates native to ion trap systems."""
+from typing import (
+    Union, Tuple, Optional, List, Callable, cast, Iterable, Sequence,
+)
+
+import numpy as np
+
+from cirq import value, protocols
+from cirq.ops import gate_features, eigen_gate
+
+class CxNOTGate(eigen_gate.EigenGate,
```

Comment:
Rename to `XXGate`. I guess we should also add ZZ and YY while we're at it. Can be done after.